edge_multi_trigger.go

```diff
@@ -355,12 +362,16 @@ func (s *EMTState) edgeMultiComputeRecordSpecs(raw []RawType, frameIndexOfraw0 F
 }
 
 func (dsp *DataStreamProcessor) edgeMultiTriggerComputeAppend(records []*DataRecord) []*DataRecord {
-	segment := &dsp.stream.DataSegment
-	recordSpecs := dsp.EMTState.edgeMultiComputeRecordSpecs(segment.rawData, segment.firstFrameIndex)
+	stream := dsp.stream
+	// fmt.Println("firstFrameIndex", stream.firstFrameIndex, "len(segment.rawData)", len(stream.rawData))
+	recordSpecs := dsp.EMTState.edgeMultiComputeRecordSpecs(stream.rawData, stream.firstFrameIndex)
 	for _, recordSpec := range recordSpecs {
-		record := dsp.triggerAtSpecificSamples(segment, int(recordSpec.firstRisingFrameIndex-segment.firstFrameIndex), int(recordSpec.npre), int(recordSpec.nsamp))
+		record := dsp.triggerAtSpecificSamples(&stream, int(recordSpec.firstRisingFrameIndex-stream.firstFrameIndex), int(recordSpec.npre), int(recordSpec.nsamp))
 		records = append(records, record)
 	}
-	dsp.stream.TrimKeepingN(dsp.EMTState.NToKeepOnTrim()) // see comment at end of edgeMultiComputeAppendRecordSpecs
+	// n_before := len(stream.rawData)
+	// n_keep := dsp.EMTState.NToKeepOnTrim()
+	// dsp.stream.TrimKeepingN(n_keep) // see comment at end of edgeMultiComputeAppendRecordSpecs
```

Comment:
This is a `TrimKeepingN` instance that you have removed. Can you explain the thinking? Was edge-multi trimming streams prematurely? How did this ever work...? Did this make EMT play badly with all other triggers (as noted on the Dcom GUI)?

Reply:
Actually, I think my problem reviewing this PR is that I don't know what problem you're trying to fix...??

Reply:
The issue is dastard would crash every time i started edge multi triggers. I have no idea how it ever worked.

Reply:
Okay, so this fixes #279 (I just added that as an issue).